File: python/strjoin/format.py
s1 = 'Hello'
s2 = '{}, World!'.format(s1)
print(s2)
# Hello, World!


s1 = 'Hello'
s2 = 'World'
s3 = '{}, {}!'.format(s1, s2)
print(s3)
# Hello, World!


s1 = 'Hello'
s2 = 'World'
s3 = '{one}, {two}!'.format(one=s1, two=s2)
print(s3)
# Hello, World!


s1 = 'Hello'
s2 = 'World'
s3 = '{one}, {two}!'.format(two=s2, one=s1)
print(s3)
# Hello, World!

File: c/printfstring/format.c
#include <stdio.h>

int main(void) {
    // 表示桁数の調整
    printf("[%20s]\n", "Hello, World!");  // [       Hello, World!]
    printf("[%.5s]\n", "Hello, World!");  // [Hello]

    // 右詰め・左詰めの指定
    printf("[%-20s]\n", "Hello, World!");  // [Hello, World!       ]
    printf("[%+20s]\n", "Hello, World!");  // [       Hello, World!]

    return 0;
}
